citest/json_predicate/binary_predicate.py

```diff
@@ -177,6 +182,38 @@ def _verify_elem(self, context, elem, the_list):
     return False
 
 
+class ListSimilarPredicate(_BaseListMembershipPredicate):
```

Comment:
Relevant to the port?

Reply:
actually yes, but probably should have done that in a separate PR.
In python 3 you cannot compare dictionaries. We used to use that before to compare list membership by first sorting to it would be easy to compare. Now I cannot figure out how to do something efficient so I'm using the subset check both ways to test a <=b and b <=a because the subset check has logic in it.

Reply:
I have it in this PR because it is tied to python3 since that's the only reason for the change.